addons/mail/static/src/js/services/mail_document_thread_manager.js

```diff
@@ -213,13 +228,22 @@ MailManager.include({
      * @param {StorageEvent} ev
      */
     _onStorage: function (ev) {
-        if (ev.key === this.DOCUMENT_THREAD_STATE_KEY) {
-            var state = this.call('local_storage', 'getItem', this.DOCUMENT_THREAD_STATE_KEY);
-            this._updateDocumentThreadWindows(state);
+        var value = JSON.parse(ev.newValue);
+        if (ev.key.indexOf(this.DOCUMENT_THREAD_STATE_KEY) !== -1) {
```

Comment:
You can do the test with tabId just after value parsing so it's not written twice and I think the function will be clearer.
```suggestion
        if (value.tabId === this._tabId) {
             if (ev.key.indexOf(this.DOCUMENT_THREAD_STATE_KEY) !== -1) {
                 (...)
             } else if (ev.key === this.DOCUMENT_THREAD_MESSAGE_KEY) {
                 (...)
             }
        }
```

Reply:
Not all JSON parseable data from local storage are objects with `tabId`, hence each valid document thread related data should make their check.

Reply:
I see :)